samples/notifications.js

```diff
@@ -0,0 +1,198 @@
+/**
```

Comment:
This is awesome, but @lukesneeringer @jmdobry -- are we supposed to be writing and testing these `samples/` files now?

Reply:
Short answer: yes.

Long answer: Yes, because it's better for the samples in the library's API ref docs to be tested, so let's test them. In order to easily test them, they need to be in real .js files in the `samples/` directory. In order for the samples to be importable into the API ref docs, they'll need region tags. We have a code sample styleguide, which I'll send you. Please add me and/or @ace-n as reviewers on samples. One thing to be extra careful of is changing or deleting samples that are imported somewhere on cloud.google.com besides the API ref docs, so it's best to leave that to me or @ace-n.

A nice thing about writing system tests for the samples is that those tests serve as dual-role as system tests for the library.

Reply:
IIUC, the output for the `samples/*.js` files is the cloud.google.com site. Those examples are supported by text describing use cases, and consistent between languages. Previously, I've understood those to be determined by "docs writers" (official title unknown) working side-by-side with API owners. It feels backwards, and potentially time/effort-wasting, for us to guess what those use case snippets should be.

At the very least, I think we need a more clear explanation, possibly in CONTRIBUTING.md, about the development process given this new integration. Maybe something like:

- Who requested this new feature?
  - GCS team: They will supply us a list of region tags and pseudo code for examples they need
  - A user/us: We will alert @jmdobry/@ace-n to determine the examples we need, if any

In both cases, either @callmehiphop or I would write the examples, assuming @lukesneeringer wants this work put on us. But the benefit is, we wouldn't be instructing upwards to the docs site, the use cases they will show.

Let me know if I have an incorrect understanding of how `samples/*.js` files are being used.

Reply:
Me and @ace-n will still take care of any samples that are to be shown in the docs on cloud.google.com. The expectation is that samples previously written by you and @callmehiphop will move from untested, inline JSDoc examples to tested samples in `samples/*.js`, and going forward new samples you'll write for the API ref docs that previously would have been those inline JSDoc examples will now be located in `samples/*.js`. What @callmehiphop did in this PR is perfect. I expect you'll continue to write the same amount of samples you've been writing previously, they'll just be in a different location (with tests).

If, within `samples/*.js` you would like to segregate samples shown in cloud.google.com/storage/docs from samples written by you for the API ref docs, then that's fine.

Reply:
Thanks, I was not making the connection that we would stop writing inline examples. @callmehiphop still made new snippets in this PR-- should rhese be removed?

And yes, I think it would be a good idea to separate our samples from the others.

Reply:
> should these be removed?

I would say yes. The value of such barebones examples is reduced now that we're properly documenting the callback and promise resolution signatures.

I'll come up with a way to separate the samples.

Reply:
I forgot about the reason we didn't move the code blocks out in the first place-- so we and our contributors wouldn't forget to update them. Was this considered when it was decided to remove the inline examples? Maybe there's a way to have both, if more testing of the `@example` blocks was important. Worth noting, we did have smoke tests for the inline examples.

Reply:
@callmehiphop I'm still trying to follow along with the new snippets/samples process here, but I think @jmdobry would like to us to remove the inline `@example`s and add them as samples in separate files.

Reply:
We can probably worry about that in a different PR.

Reply:
I would lean towards solving it before we commit "to be removed" code.